### ISSUE TITLE
feat(core): add section option to TBL-001 for scoping tables by heading

### ISSUE THREAD

**Issue by nozomi-koborinai** [nozomi-koborinai/contextlint#23]
## Problem

A single Markdown file often contains multiple tables of different types. For example, `dna_management.md` has:

- A requirements table (`## What`) with columns: ID, Requirement, Stability, Notes
- A file types table (`### DNA File Types`) with columns: File Type, Naming Convention, Content
- A stability model table (`### Stability Model`) with columns: Stability, Meaning, Cost of Change

TBL-001 (`requiredColumns`) currently checks **all** tables in a file. If you set `requiredColumns: ["ID", "Stability"]`, unrelated tables (e.g., "DNA File Types") are flagged for missing "ID" column.

The `files` option (added in #16) filters by file path, but cannot distinguish between multiple tables **within the same file**.

## Affected rules

- **TBL-001** is the primary problem — it reports missing columns on every table
- TBL-002/003/004 already skip tables that don't have the target column, so they are not affected

## Proposed solution

Add an optional `section` option to TBL-001 that scopes the rule to tables under a specific heading.

```jsonc
{ "rule": "tbl001", "options": { "requiredColumns": ["ID", "Stability"], "section": "What", "files": "**/requirements.md" } }
```

### Behavior

- A "section" = everything from the matched heading to the next heading of the **same or higher level**
- The match should work on any heading level (`#`, `##`, `###`, etc.)
- Tables anywhere within the section are included (not just immediately after the heading)
- Partial match or exact match TBD (e.g., `"What"` matching `## What (Requirements)`)
- If `section` is omitted, all tables are checked (backwards compatible)

### Example

```markdown
## What (Requirements)

Some explanation text here...

| ID | Requirement | Stability | Notes |
|----|-------------|-----------|-------|
| STD-DNA-01 | ... | stable | |

## Spec (Specification)

### DNS File Types
| File Type | Naming Convention | Content |
|...|...|...|

### Stability Model
| Stability | Meaning | Cost of Change |
|...|...|...|
```

With `section: "What"`, only the first table is checked for required columns.

## Future considerations

- This option could be extended to other TBL rules if needed
- Consider whether to support nested section paths (e.g., `section: "Spec > Stability Model"`) or just simple heading text matching
